modules/dnn/include/opencv2/dnn/dnn.hpp

```diff
@@ -55,6 +55,23 @@ namespace dnn //! This namespace is used for dnn module functionlaity.
 
     typedef std::vector<int> MatShape;
 
+    /**
+     * @brief Enum of computation backends supported by layers.
+     */
+    enum Backend
+    {
+        DEFAULT,
+        HALIDE
+    };
+
+    /**
+     * @brief Compilation targets of Halide code.
+     */
+    enum HalideTarget
+    {
+        HALIDE_CPU
```

Comment:
again, let's use DNN_TARGET_CPU, DNN_TARGET_GPU etc. I would not use the word "Halide" here, because other backends, like OpenVX or even the default one, can be potentially customized to use CPU, GPU or other device to run the networks